cannot compile with clang (missing limits include)

When trying to compile with clang, `E57XmlParser.cpp` will be compiled without the required `<limits>` header:
```
/run/build/CloudCompare/plugins/core/IO/qE57IO/extern/libE57Format/src/E57XmlParser.cpp: In member function ‘virtual XMLSize_t E57FileInputStream::readBytes(XMLByte*, XMLSize_t)’:
/run/build/CloudCompare/plugins/core/IO/qE57IO/extern/libE57Format/src/E57XmlParser.cpp:143:37: error: ‘numeric_limits’ is not a member of ‘std’
  143 |       const int64_t size_max = std::numeric_limits<size_t>::max();
      |                                     ^~~~~~~~~~~~~~
```

This header is only included for GCC>11:
https://github.com/asmaloney/libE57Format/blob/dd7d7d711fa8f6de7017c3f2b2d0b714f1db4908/src/E57XmlParser.cpp#L45-L47

Is there a reason why this header should not be included for other compiler and versions?

--- a/src/E57XmlParser.cpp
+++ b/src/E57XmlParser.cpp
@@ -25,6 +25,8 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#include <limits>
+
 #include <xercesc/sax2/Attributes.hpp>
 #include <xercesc/sax2/XMLReaderFactory.hpp>
 
@@ -41,10 +43,6 @@
 #include "ScaledIntegerNodeImpl.h"
 #include "StringNodeImpl.h"
 #include "VectorNodeImpl.h"
-
-#if __GNUC__ >= 11
-#include <limits>
-#endif
 
 using namespace e57;
 using namespace XERCES_CPP_NAMESPACE;

